crates/ra_hir_def/src/nameres/path_resolution.rs

```diff
@@ -212,14 +226,30 @@ impl CrateDefMap {
                 }
             };
         }
-        ResolvePathResult::with(curr_per_ns, ReachedFixedPoint::Yes, None)
+        return ResolvePathResult::with(curr_per_ns, ReachedFixedPoint::Yes, None);
+
+        // if it is not the last segment, we prefer builtin as module
```

Comment:
that comment seems the wrong way around, if you mean 'we prefer the builtin to the module'